test/tools/yulopti.cpp

```diff
@@ -162,6 +162,36 @@ class YulOpti
 		}
 	}
 
+	int runSteps(string source, string steps)
+	{
+		if (!parse(source))
+			return 1;
+
+		set<YulString> reservedIdentifiers;
+		*m_ast = std::get<yul::Block>(Disambiguator(m_dialect, *m_analysisInfo)(*m_ast));
```

Comment:
Can you try to reduce code duplication? I mean in the end, can't we even re-use the `runInteractive` function:

```
void runInteractive(string source, optional<string> _steps)
{
    // ...
    char option;
    if (_steps)
    {
        if (_steps->empty()) return;
        option = _steps.front();
        _steps = steps.substr(1);
    }
    else
        option = static_cast<char>(readStandardInputChar());
    // ...

}
```
Or don't you want the intermediate outputs to be printed?

Reply:
I want to neither print the input nor the intermediate outputs nor the option descriptions - it's meant for non-interactive runs, i.e. the result should be easy to parse and if I want intermediate outputs, I can just run the steps in separate invocations...

Also the disambiguation logic differs slightly, since I don't need to do it over and over again, since I don't allow the var name cleaner... i.e. one global context is enough and I don't need to recreate one each step...

But yeah, it could still be merged together... but then it's a lot of conditionals replacing a bit of duplication...

Reply:
I could probably do the opposite, i.e. move all logic for running a single step into a function and call it from ``runInteractive`` each time...

Reply:
I think that at least the part between `parse()` and context creation looks like it could be extracted into a function. And the rest could be replaced with a call to `OptimiserSuite::runSequence()` as I described above.